### ISSUE TITLE
WavPack 32 bit float point to Flac - program crashes with an error: No program define

### ISSUE THREAD

**Issue by newchipo** [flacon/flacon#204]
Hi! When I try convert WapPack file 32-bit floats at 192000 Hz to Flac, Flacon show me error. I think this is because flac does not know how to work with 32-bit files. The solution to the problem may be the conversion of such files via ffmpeg. With a warning that a 32-bit file will become 24 bits: `ffmpeg -i input_file.wv output_file.flac`

```
2023.05.26 07:45:53.272 +04 [53349] Splitter: debug: extract:  "/home/denis/Temp/music/birtman/Биртман - Подбухну (LP).wv"  [ "00:00:00" : "02:54:05" ] OUT: "/home/denis/Temp/music/birtman/Биртман/2020 - Подбухну (LP)/tmp.dizUPo/track-0001_01.wav"
2023.05.26 07:45:54.905 +04 [53349] Splitter: debug: Splitter trackReady:  Track { trackNum: 1 trackCount:12 diskId:"" Artist:"Биртман" Album:"Подбухну (LP)" Title:"Рэп Культура"} "/home/denis/Temp/music/birtman/Биртман/2020 - Подбухну (LP)/tmp.dizUPo/track-0001_01.wav"
2023.05.26 07:45:54.905 +04 [53349] Splitter: debug: extract:  "/home/denis/Temp/music/birtman/Биртман - Подбухну (LP).wv"  [ "02:54:05" : "06:23:44" ] OUT: "/home/denis/Temp/music/birtman/Биртман/2020 - Подбухну (LP)/tmp.dizUPo/track-0001_02.wav"
2023.05.26 07:45:54.906 +04 [53351] Encoder: debug: Start encoder: "/usr/bin/flac --force --silent --compression-level-5 - -o '/home/denis/Temp/music/birtman/Биртман/2020 - Подбухну (LP)/tmp.dizUPo/track-0001_01.encoded.flac'"
2023.05.26 07:45:54.906 +04 [53351] Encoder: debug: Input audio: bitsPerSample = 32  sampleRate = 192000
2023.05.26 07:45:54.906 +04 [53351] Encoder: debug: Required:    bitsPerSample = 24  sampleRate = 192000
2023.05.26 07:45:54.906 +04 [53351] Encoder: debug: Start resampler: " --type wav - -b 24 --type wav - rate -v 192000"
2023.05.26 07:45:54.906 +04 [53351] Encoder: debug: DeEmphasis is not required
2023.05.26 07:45:54.906 +04 [53351] ExtProgram: warning: ERROR
2023.05.26 07:45:54.906 +04 [53351] ExtProgram: warning: "resampler: The '' program crashes"
2023.05.26 07:45:54.906 +04 [53351] ExtProgram: warning: Program with args: " --type wav - -b 24 --type wav - rate -v 192000"
2023.05.26 07:45:54.906 +04 [53351] ExtProgram: warning: Error: QProcess::FailedToStart
2023.05.26 07:45:54.906 +04 [53351] ExtProgram: warning: Error string: "No program defined"
2023.05.26 07:45:54.928 +04 [53328] default: critical: "Track 1. Encoder error:The '' program crashes with an error: No program defined"
```

**Comment by SokoloffA** [comment]
@newchipo 
TLDR It looks like you don't have the sox program installed. 

In your case, Flacon will reduce the rate/bitsdepth to an acceptable for out format. This is what the sox program is for. It looks like you don't have it installed or not specified in the Flacon settings.

Another question is why my program didn't tell you about the lack of sox. Email me at sokoloff.a@gmail.com, I have some questions.

**Comment by SokoloffA** [comment]
I found and fixed the problem. In your situation the program did not check whether the sox program was installed.

If you have time, please check the new version from [AppImage](https://github.com/flacon/flacon/suites/13190209130/artifacts/718101460).

**Comment by SokoloffA** [comment]
Included in release 11.1.0
